### ISSUE TITLE
Add an autoconf template

### ISSUE THREAD

**Issue by atheriel** [r-lib/usethis#313]
Note: I actually implemented this functionality before I read the request to file an issue beforehand in the contributor guidelines, so if you'd like to see an implementation you can check out [this commit](https://github.com/atheriel/usethis/commit/279e4b2413bf4d1400e7a0449fc18d24be2ca36c).

One seriously difficult hurdle I have encountered as a package author is determining how to add `autoconf`-based configure scripts to an Rcpp package that needed to detect a particular C++ library installed on a system. This is the recommended approach in [*Writing R Extensions*](https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Configure-example) for solving this problem, but I found few accessible resources on how to actually write these scripts (specific to R, anyway). Moreover, there are actually a ton of other, related steps in order to get R CMD check to play nicely with `autoconf`'s generated files (permissions, the `cleanup` script, `autoconf`-related artifacts, etc).

All this to say that once I finally figured it out I wanted to make it easier for others, and the **usethis** package seems like the most promising veneue for providing "this is how you should do this" functionality, even though this is a relatively uncommon need for package authors.

In short, I imagine a`use_autoconf()` function that:

- generates a simple `configure.ac` file with some helpful comments to get you started on what you might want to modify;
- generates the `cleanup` script used by R to remove `autoconf`-related build artifacts;
- generates (or modifies your existing) `src/Makevars` to work nicely with `autoconf`;
- adds various artifacts to the ignore files; and
- adds a git commit hook to remind you to run `autoconf` itself when you configure script is out-of-date

Is this outside of the scope of the **usethis** package?

**Comment by atheriel** [comment]
That’s understandable. Thanks for your work on the package.
